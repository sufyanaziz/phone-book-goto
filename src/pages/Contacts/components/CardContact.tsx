import React from "react";
import { Theme, css } from "@emotion/react";
import Text from "@common/components/Text";
import Button from "@common/components/Button";

type TCardContactData = {
  id: number;
  first_name: string;
  last_name: string;
  phones: { number: string }[];
};

type TCardContact = {
  data: TCardContactData;
  onDelete: (data: TCardContactData) => void;
  onDetail: (data: TCardContactData) => void;
  onEdit: (data: TCardContactData) => void;
};

const CardContact: React.FC<TCardContact> = ({ data, ...rest }) => {
  const cardContentStyle = (theme: Theme) => {
    return css({
      border: `1px solid black`,
      borderBottom: 0,
      padding: `${theme.padding[4]} ${theme.padding[3]}`,
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
      ":last-child": {
        marginBottom: 0,
        borderBottom: "1px solid black",
      },
      ".card-content-action": {
        display: "flex",
        gap: "4px",
      },
    });
  };
  return (
    <div css={cardContentStyle}>
      <div className="card-content-info">
        <Text
          text={`${data.first_name} ${data?.last_name || ""}`}
          type="bold"
        />
        <Text text={data.phones?.[0]?.number} />
      </div>
      <div className="card-content-action">
        <Button
          label="Detail"
          type="default"
          onClick={() => rest.onDetail(data)}
        />
        <Button label="Edit" type="warning" onClick={() => rest.onEdit(data)} />
        <Button
          label="Delete"
          type="important"
          onClick={() => rest.onDelete(data)}
        />
      </div>
    </div>
  );
};

export default CardContact;
