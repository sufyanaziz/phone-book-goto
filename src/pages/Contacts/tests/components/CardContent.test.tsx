import React from "react";
import renderer from "react-test-renderer";
import CardContact from "@pages/Contacts/components/CardContact";

describe("Component CardContact", () => {
  it("should render CardContact component correctly", () => {
    const component = renderer.create(
      <CardContact
        data={{
          id: 123,
          first_name: "dummy first name",
          last_name: "dummy last name",
          phones: [{ number: "8767322" }],
        }}
        onDelete={jest.fn()}
        onDetail={jest.fn()}
        onEdit={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
