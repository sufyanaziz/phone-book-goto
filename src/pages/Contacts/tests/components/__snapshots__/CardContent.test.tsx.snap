// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CardContact should render CardContact component correctly 1`] = `
<div
  css={[Function]}
>
  <div
    className="card-content-info"
  >
    <div
      className=""
      css={[Function]}
    >
      dummy first name dummy last name
    </div>
    <div
      className=""
      css={[Function]}
    >
      8767322
    </div>
  </div>
  <div
    className="card-content-action"
  >
    <button
      className="default"
      css={[Function]}
      onClick={[Function]}
    >
      Detail
    </button>
    <button
      className="warning"
      css={[Function]}
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="important"
      css={[Function]}
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
